### PR TITLE
fix: report all validations failed at once

### PR DESCRIPTION
**Pull request by Xstoudi**
## Proposed changes

See #20 

## Types of changes
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/adonisjs/env/blob/master/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

File: src/Env/index.ts [hunk 1]
```diff
@@ -96,9 +98,15 @@ export class Env implements EnvContract {
...
+		const errors: string[] = []
 		Object.keys(this.validationSchema).forEach((key) => {
-			this.envCache[key] = this.validationSchema[key](key, this.get(key))
+			try {
+				this.envCache[key] = this.validationSchema[key](key, this.get(key))
+			} catch (err) {
+				errors.push((err as Exception).message)
+			}
 		})
+		if (errors.length > 0) throw new Exception(errors.join(EOL), 500)
```
Comment (targos) [inline src/Env/index.ts:109]:
```suggestion
		if (errors.length > 0) throw new Exception(errors.join('\n'), 500)
```

I don't think anyone would need the separator to be `\r\n` on Windows. If you take `error.stack` for example, V8 always sets it to `\n`.

**Comment by thetutlage** [comment]
The issue here is that the exception codes are not useful anymore. We are just concatenating the messages and displaying them. 

I am not sure what is the right answer to this problem. But exception codes are something I want to keep, as they will be documented on the website

**Comment by Xstoudi** [comment]
Were they useful before? This part run before http server starts.

Alternatively, I could collect both error types (missing and invalid) as separate arrays. It won't be optimal for the user, but having to start the server two times instead of N is still an improvement.

What do you think?

**Comment by targos** [comment]
waiting for feedback

**Pull request closed without merging**